code/modules/mob/living/carbon/diona_base.dm

```diff
@@ -61,14 +61,17 @@ var/list/diona_banned_languages = list(
 /mob/living/carbon/proc/diona_handle_radiation(var/datum/dionastats/DS)
 	//Converts radiation to stored energy if its needed, and gives messages related to radiation
 	//Rads can be used to heal in place of light energy, that is handled in the regular regeneration proc
+	if(DS.regening_organ)
+		return
 
-	var/radiation = total_radiation
-
-	if (radiation && DS.stored_energy < (DS.max_energy * 0.8)) //Radiation can provide energy in place of light
-		radiation -= 2
+	if (total_radiation && DS.stored_energy < (DS.max_energy * 0.8)) //Radiation can provide energy in place of light
+		apply_radiation(-2)
 		DS.stored_energy += 2
 
-	radiation -= 0.5 //Radiation is gradually wasted if its not used for something
+	
+	apply_radiation(-0.5) //Radiation is gradually wasted if its not used for something
+	if(total_radiation < 0)
+		total_radiation = null
```

Comment:
Shouldn't this be `0`?

Reply:
Not really. Apparently default total_radiation is `null`. Idk why